middleware/locale/locale.go

```diff
@@ -0,0 +1,61 @@
+package locale
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/mholt/caddy/middleware"
+	"github.com/mholt/caddy/middleware/locale/method"
+)
+
+// Locale is a middleware to detect the user's locale.
+type Locale struct {
+	Next             middleware.Handler
+	AvailableLocales []string
+	Methods          []method.Method
+	PathScope        string
+	Configuration    *method.Configuration
+}
+
+// ServeHTTP implements the middleware.Handler interface.
+func (l *Locale) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
+	if !middleware.Path(r.URL.Path).Matches(l.PathScope) {
+		return l.Next.ServeHTTP(w, r)
+	}
```

Comment:
And to note, this is a very beautifully written ServeHTTP method as-is; it's quite elegant. There should just be some clarification on the scoping in the docs, or the logic should be updated.